rbac/core/kafka_consumer.py

```diff
@@ -630,19 +634,56 @@ def _process_relations_message(self, debezium_msg: DebeziumMessage) -> bool:
                 messages_processed_total.labels(message_type="relations", status="validation_failed").inc()
                 return False
 
+            # Extract the org_id from resource context
+            resource_context = debezium_msg.payload.get("resource_context")
+            org_id = resource_context["org_id"]
+
             # Create structured replication message
             replication_msg = ReplicationMessage.from_payload(debezium_msg.payload)
 
             logger.info(
-                f"Processing relations message - aggregateid: {debezium_msg.aggregateid}, "
+                f"Processing relations message - org_id: {org_id}, "
+                f"aggregateid: {debezium_msg.aggregateid}, "
                 f"event_type: {debezium_msg.event_type}, "
                 f"relations_to_add: {len(replication_msg.relations_to_add)}, "
                 f"relations_to_remove: {len(replication_msg.relations_to_remove)}"
             )
 
-            # TODO: Add actual processing logic here
-            # This is where you would integrate with the relation replication system
+            # Get tenant by org_id
+            try:
+                tenant = Tenant.objects.get(org_id=org_id)
+            except Tenant.DoesNotExist:
+                logger.error(f"Tenant not found for org_id: {org_id}")
+                messages_processed_total.labels(message_type="relations", status="tenant_not_found").inc()
+                return False
+
+            # Do tuple deletes for relationships
+            replication_delete_response = relations_api_replication._delete_relationships(
+                relationships=replication_msg.relations_to_remove
+            )
+
+            # Do tuple writes for relationships
+            replication_add_response = relations_api_replication._write_relationships(
+                relationships=replication_msg.relations_to_add
+            )
+
+            # Extract consistency token from responses
+            token = getattr(replication_add_response.consistency_token, "token", None) or getattr(
+                replication_delete_response.consistency_token, "token", None
+            )
+
+            if token:
```

Comment:
you can do `if token and org_id`: and  get tenant only if this condition is true ?